### PR TITLE
cmake: Add BOND_ENABLE_JAVA (default=False).

### PR DESCRIPTION
**Pull request by tstein**
Intended to resolve https://github.com/Microsoft/bond/issues/691.

### CODE REVIEW

File: cmake/Config.cmake [hunk 1]
```diff
@@ -209,6 +209,10 @@ set (BOND_ENABLE_COMM
     "FALSE"
     CACHE BOOL "If FALSE, then do not build Comm")
 
+set (BOND_ENABLE_JAVA
```
Comment (chwarr) [inline cmake/Config.cmake]: The `GRADLE_FOUND` and `MAVEN_FOUND` variables in FindJavaBuildTools.cmake don't appear to be used. Can they be removed?

Comment (chwarr) [inline cmake/Config.cmake]: Can we make FindJavaBuildTools.cmake fail now, as they'll only be invoked if BOND_ENABLE_JAVA is true?

Comment (tstein) [reply cmake/Config.cmake]: Done.

Comment (tstein) [reply cmake/Config.cmake]: Done.

**Pull request merged**